### ISSUE TITLE
Error on npm install

### ISSUE THREAD

**Issue by MarcGodard** [bitovi-components/bit-c3#30]
`node -v`
v7.4.0

```
Marcs-MacBook-Pro:site marcgodard$ npm install bit-c3 --save

> contextify@0.1.15 install /Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/contextify
> node-gyp rebuild

  CXX(target) Release/obj.target/contextify/src/contextify.o
../src/contextify.cc:131:56: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                                       ^
/Users/marcgodard/.node-gyp/7.4.0/include/node/v8.h:3292:52: note: 'NewInstance' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version", Local<Object> NewInstance() const);
                                                   ^
../src/contextify.cc:150:16: error: no member named 'SetAccessCheckCallbacks' in 'v8::ObjectTemplate'
        otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
        ~~~~~  ^
../src/contextify.cc:182:51: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                                                  ^
/Users/marcgodard/.node-gyp/7.4.0/include/node/v8.h:2949:30: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
                Local<Value> GetRealNamedProperty(Local<String> key));
                             ^
../src/contextify.cc:209:38: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
        if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
                                     ^
/Users/marcgodard/.node-gyp/7.4.0/include/node/v8.h:2949:30: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
                Local<Value> GetRealNamedProperty(Local<String> key));
                             ^
../src/contextify.cc:210:42: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
            !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
                                         ^
/Users/marcgodard/.node-gyp/7.4.0/include/node/v8.h:2949:30: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
                Local<Value> GetRealNamedProperty(Local<String> key));
                             ^
4 warnings and 1 error generated.
make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:191:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Darwin 16.4.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/contextify
gyp ERR! node -v v7.4.0
gyp ERR! node-gyp -v v3.4.0
gyp ERR! not ok 
easyapp@0.0.0 /Users/marcgodard/CloudStation/Projects/YOURICO/site
└── cssify@0.6.0  extraneous

npm WARN easyapp@0.0.0 No repository field.
npm WARN easyapp@0.0.0 No license field.
npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "bit-c3" "--save"
npm ERR! node v7.4.0
npm ERR! npm  v4.0.5
npm ERR! code ELIFECYCLE

npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.15 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the contextify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs contextify
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls contextify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/marcgodard/CloudStation/Projects/YOURICO/site/npm-debug.log
```

**Comment by MarcGodard** [comment]
Looks like I need an older version of node.

**Comment by matthewp** [comment]
If you were to create a blank project (without donejs or anything) and npm install bit-c3 would you still get this error?

**Comment by MarcGodard** [comment]
Yes, still get this error.

I changed node version to pre 7 (6.95 or something) and I still got warnings, but no errors.

However, now when I go to use bit-c3 I get the following error:
```
Marcs-MacBook-Pro:site marcgodard$ donejs develop

> easyapp@0.0.0 develop /Users/marcgodard/CloudStation/Projects/XXX/site
> done-serve --develop --port 8080

done-serve starting on http://localhost:8080
Live-reload server listening on port 8012
Potentially unhandled rejection [6] TypeError: Error loading "d3@3.5.0#d3" at file:/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/d3/d3.js
Error loading "d3@3.5.0#d3" from "easyapp@0.0.0#index.stache!done-autorender@0.8.0#autorender" at file:/Users/marcgodard/CloudStation/Projects/XXX/site/src/index.stache
Error evaluating file:/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/d3/d3.js
Cannot read property 'prototype' of undefined
    at file:/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/d3/d3.js:24:49
    at file:/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/d3/d3.js:9426:2
    at ContextifyScript.Script.runInThisContext (vm.js:25:33)
    at Object.exports.runInThisContext (vm.js:77:17)
    at doEval (/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/steal-systemjs/dist/system.src.js:2059:10)
    at __eval (/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/steal-systemjs/dist/system.src.js:1976:7)
    at Loader.exec [as __exec] (/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/steal-systemjs/dist/system.src.js:339:5)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/steal-systemjs/dist/system.src.js:1517:16)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/steal-systemjs/dist/system.src.js:1843:32)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/XXX/site/node_modules/steal/main.js:599:56)
```

**Comment by MarcGodard** [comment]
I tried to update c3 to version 4.11 (instead of 4.10) because of bug fixes, and updated d3 to 3.5.17 (instead of 3.5) again because that is the last version before 4 and it would have bug fixes.

These things didn't make a difference.

**Comment by phillipskevin** [comment]
With the dependencies installed with Node 7, can you run `npm ls contextify` to see where that dependency is coming from?

**Comment by MarcGodard** [comment]
Not sure what you mean. I went back to a newer version of node because of that error with d3 that I couldn't resolve.

Bellow you will find the install, then the command you asked to be run, which returned nothing.
```
Marcs-MacBook-Pro:site marcgodard$ npm install bit-c3

> contextify@0.1.15 install /Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/contextify
> node-gyp rebuild

  CXX(target) Release/obj.target/contextify/src/contextify.o
../src/contextify.cc:131:56: warning: 'NewInstance' is deprecated [-Wdeprecated-declarations]
        Local<Object> wrapper = Nan::New(constructor)->NewInstance();
                                                       ^
/Users/marcgodard/.node-gyp/7.5.0/include/node/v8.h:3292:52: note: 'NewInstance' has been explicitly marked deprecated here
  V8_DEPRECATED("Use maybe version", Local<Object> NewInstance() const);
                                                   ^
../src/contextify.cc:150:16: error: no member named 'SetAccessCheckCallbacks' in 'v8::ObjectTemplate'
        otmpl->SetAccessCheckCallbacks(GlobalPropertyNamedAccessCheck,
        ~~~~~  ^
../src/contextify.cc:182:51: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
        Local<Value> rv = Nan::New(ctx->sandbox)->GetRealNamedProperty(property);
                                                  ^
/Users/marcgodard/.node-gyp/7.5.0/include/node/v8.h:2949:30: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
                Local<Value> GetRealNamedProperty(Local<String> key));
                             ^
../src/contextify.cc:209:38: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
        if (!Nan::New(ctx->sandbox)->GetRealNamedProperty(property).IsEmpty() ||
                                     ^
/Users/marcgodard/.node-gyp/7.5.0/include/node/v8.h:2949:30: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
                Local<Value> GetRealNamedProperty(Local<String> key));
                             ^
../src/contextify.cc:210:42: warning: 'GetRealNamedProperty' is deprecated [-Wdeprecated-declarations]
            !Nan::New(ctx->proxyGlobal)->GetRealNamedProperty(property).IsEmpty()) {
                                         ^
/Users/marcgodard/.node-gyp/7.5.0/include/node/v8.h:2949:30: note: 'GetRealNamedProperty' has been explicitly marked deprecated here
                Local<Value> GetRealNamedProperty(Local<String> key));
                             ^
4 warnings and 1 error generated.
make: *** [Release/obj.target/contextify/src/contextify.o] Error 1
gyp ERR! build error 
gyp ERR! stack Error: `make` failed with exit code: 2
gyp ERR! stack     at ChildProcess.onExit (/usr/local/lib/node_modules/npm/node_modules/node-gyp/lib/build.js:276:23)
gyp ERR! stack     at emitTwo (events.js:106:13)
gyp ERR! stack     at ChildProcess.emit (events.js:192:7)
gyp ERR! stack     at Process.ChildProcess._handle.onexit (internal/child_process.js:215:12)
gyp ERR! System Darwin 16.4.0
gyp ERR! command "/usr/local/bin/node" "/usr/local/lib/node_modules/npm/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
gyp ERR! cwd /Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/contextify
gyp ERR! node -v v7.5.0
gyp ERR! node-gyp -v v3.5.0
gyp ERR! not ok 
easyapp@0.0.0 /Users/marcgodard/CloudStation/Projects/YOURICO/site
└── cssify@0.6.0  extraneous

npm WARN easyapp@0.0.0 No repository field.
npm WARN easyapp@0.0.0 No license field.
npm ERR! Darwin 16.4.0
npm ERR! argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "bit-c3"
npm ERR! node v7.5.0
npm ERR! npm  v4.1.2
npm ERR! code ELIFECYCLE

npm ERR! contextify@0.1.15 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the contextify@0.1.15 install script 'node-gyp rebuild'.
npm ERR! Make sure you have the latest version of node.js and npm installed.
npm ERR! If you do, this is most likely a problem with the contextify package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get information on how to open an issue for this project with:
npm ERR!     npm bugs contextify
npm ERR! Or if that isn't available, you can get their info via:
npm ERR!     npm owner ls contextify
npm ERR! There is likely additional logging output above.

npm ERR! Please include the following file with any support request:
npm ERR!     /Users/marcgodard/CloudStation/Projects/YOURICO/site/npm-debug.log
Marcs-MacBook-Pro:site marcgodard$ npm ls contextify
easyapp@0.0.0 /Users/marcgodard/CloudStation/Projects/YOURICO/site
└── (empty)

Marcs-MacBook-Pro:site marcgodard$ 
```

**Comment by matthewp** [comment]
This is caused by the version of d3 being used. Will see if it can be upgraded.

**Comment by matthewp** [comment]
Fixed, doing a release.

**Comment by matthewp** [comment]
Not solved yet.

**Comment by matthewp** [comment]
Fixed in bit-c3 0.1.4

**Comment by MarcGodard** [comment]
Hey guys, upgraded to the newest (bit-c3 0.1.4). It is loading visually now in develop, however, I get this error in the terminal:

```
Marcs-MacBook-Pro:site marcgodard$ donejs develop

> easyapp@0.0.0 develop /Users/marcgodard/CloudStation/Projects/YOURICO/site
> done-serve --develop --port 8080

done-serve starting on http://localhost:8080
WARN: can/construct/construct.js: extending a can.Construct without calling extend
Received client connection
WARN: can/construct/construct.js: extending a can.Construct without calling extend
Potentially unhandled rejection [10] TypeError: Error loading "d3@3.5.17#d3" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js
Error loading "d3@3.5.17#d3" from "easyapp@0.0.0#components/testing.component!done-component@0.4.1#component" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/src/components/testing.component
Error loading "d3@3.5.17#d3" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js
Error loading "d3@3.5.17#d3" from "easyapp@0.0.0#components/testing.component/template" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/src/components/testing.component.template
Error evaluating file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js
Cannot read property 'prototype' of undefined
    at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js:33:46
    at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js:9554:2
    at ContextifyScript.Script.runInThisContext (vm.js:23:33)
    at Object.exports.runInThisContext (vm.js:74:17)
    at doEval (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:2059:10)
    at __eval (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:1976:7)
    at Loader.exec [as __exec] (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:339:5)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:1517:16)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:1843:32)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal/main.js:599:56)
Potentially unhandled rejection [14] TypeError: Error loading "d3@3.5.17#d3" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js
Error loading "d3@3.5.17#d3" from "easyapp@0.0.0#components/testing.component!done-component@0.4.1#component" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/src/components/testing.component
Error loading "d3@3.5.17#d3" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js
Error loading "d3@3.5.17#d3" from "easyapp@0.0.0#components/testing.component/template" at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/src/components/testing.component.template
Error evaluating file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js
Cannot read property 'prototype' of undefined
    at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js:33:46
    at file:/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/d3/d3.js:9554:2
    at ContextifyScript.Script.runInThisContext (vm.js:23:33)
    at Object.exports.runInThisContext (vm.js:74:17)
    at doEval (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:2059:10)
    at __eval (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:1976:7)
    at Loader.exec [as __exec] (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:339:5)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:1517:16)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal-systemjs/dist/system.src.js:1843:32)
    at Loader.loader.instantiate (/Users/marcgodard/CloudStation/Projects/YOURICO/site/node_modules/steal/main.js:599:56)
Live-reload server listening on port 8012
Received client connection
```

And then eventually in my console I now get this:

```
Potentially unhandled rejection [4] RangeError: Maximum call stack size exceeded
    at Object.can.isPromise (http://localhost:8080/node_modules/can/util/can.js:27:11)
    at Map.setup (http://localhost:8080/node_modules/can/list/list.js:94:13)
    at Map.setup (http://localhost:8080/node_modules/can/list/sort/sort.js:42:10)
    at Function.newInstance (http://localhost:8080/node_modules/can/construct/construct.js:197:23)
    at Map.init (http://localhost:8080/node_modules/can/construct/construct.js:510:30)
    at new Map (eval at extend (:483:4), <anonymous>:1:44)
    at Map.map (http://localhost:8080/node_modules/can/list/list.js:870:23)
    at Map.get (eval at __eval (http://localhost:8080/node_modules/steal/steal.js:2858:5), <anonymous>:143:75)
    at can.Compute._get (http://localhost:8080/node_modules/can/compute/proto_compute.js:209:19)
    at can.Compute.get (http://localhost:8080/node_modules/can/compute/proto_compute.js:321:17)
```

with some of these before it that I have not seen before:
```
The %root property cannot be updated
```
